servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/Single.java

```diff
@@ -1087,6 +1088,32 @@ public final Cancellable subscribe(Consumer<? super T> resultConsumer) {
         return new SucceededSingle<>(value);
     }
 
+    /**
+     * Creates a realized {@link Single} which always completes successfully with the result of the provided
```

Comment:
QQ.  Single.defer() has the same situation as Single.fromCallable() and Single.fromSupplier() in that there can be blocking calls or an exception thrown during Supplier.get().  Yet the Javadoc for defer() doesn't include this text, the implementation isn't using ThreadInterruptingCancellable, and the unit tests don't test for an interrupt.  What am I missing in the SingleDefer implementation?

Reply:
Good question!

There is a subtle difference between `defer()` and `fromCallable()`/`fromSupplier()`:

The provided entity in `defer()` is __providing the source__ (potentially new instance) for each subscribe call. It is typically use to convert eager/hot sources to lazy/cold. However, the provided entity in `fromCallable()`/`fromSupplier()` is __implementing the source__. We expect implementation of sources to be potentially blocking whereas the act of providing a new source is less likely to be blocking.

From an implementation point of view, `defer()` does not interact with the `Subscriber`  so there is not much it is expected to do as compared to `fromCallable()`/`fromSupplier()` which interact with the `Subscriber`. 

Hope this clarifies.

Reply:
It does, from a theoretical standpoint.  From a practical standpoint, while this would be the intended use of defer() going forward, and since Single was missing fromCallable() and fromSupplier() until now, I've seen the following code:

```
   Single.defer(() -> {
       final SomeObj result = << make blocking call >>
       return Single.succeeded(result);
   });
```

Going forward, you'd want to do that in Single.fromCallable() or fromSupplier(), of course, but I can't help but think there's a bunch of code out there already doing the above.

Reply:
Hmm ok. I think that usage is "unexpected" from a design point of view but I hear what you are saying. It may be a good idea to add the "blocking details" to `defer()` javadoc also. Do you want to do that?

To clarify; what we did for these newly added methods was just to make sure that if cancellation is done from a different thread it can interrupt the thread running `Callable#call()`. From the point of view of protecting from blocking code, we still have to offload the subscribe call, which is expected out of the users.

Reply:
I'm fine leaving the Javadoc the same if you're not concerned.